### DISCUSSION TITLE
Updating vercel.json redirects during `npm run generate` not working

### DISCUSSION THREAD

**Discussion by falko-apheris** [vercel/community#2787] [Help]
### Summary

My project uses nuxt to generate a static site. During the build process `npm run generate` I pull data from a CMS to set no-index pages and redirects. The redirects from the CMS are added to existing redirects from the `vercel.json` file. 

When I visit my page:
- redirects that were in the vercel.json file before work
- redirects added during `npm run generate` to `vercel.json` are not working.

What can I do to add redirects during build or maybe earlier? 

### Steps to Reproduce

add something like this to `nuxt.config.ts`
```ts
export default defineNuxtConfig({
	modules: [
		async (_, nuxt) => {
			const response = await fetch(`https://example.com/redirects.json`)
			const data = await response.json()

			// Update vercel.json redirects
			const vercelJson = JSON.parse(fs.readFileSync('./vercel.json', 'utf8'))
			const vercelRedirectsIndex = new Set(
				vercelJson.redirects.map(({ source }) => source)
			)
			const additionalRedirects = data.data.setting.redirects.filter(
				({ source }) => !vercelRedirectsIndex.has(source)
			)
			console.log('additionalRedirects', additionalRedirects)
			vercelJson.redirects.push(...additionalRedirects)
			fs.writeFileSync('./vercel.json', JSON.stringify(vercelJson, null, 2))
		},
	]
})
```
then run `nuxi generate`

**Comment by amyegan** [comment]
I think [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) may be a better solution for you. It allows for programatic redirects instead of being stuck with the static json file settings. There are quite a few [Edge Middleware example projects](https://vercel.com/templates?type=edge-middleware) that you can use as a starting point for your own solution. One of my favorites is the [Maintenance Page template](https://vercel.com/templates/next.js/maintenance-page) which uses [Edge Config](https://vercel.com/docs/storage/edge-config) with Middleware to allow redirect changes without a full redeploy.
